lambda/docker-runner.yaml

```diff
@@ -0,0 +1,111 @@
+# This template depends on having an AWS EFS in place which can be deployed with
+# https://github.com/Sage-Bionetworks/aws-infra/blob/master/templates/EFS/efs.yaml
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: S3Objects
+Description: >-
+  Docker runner product
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Container Image
+        Parameters:
+          - ImageUri
+          - EntryPoint
+          - Command
+          - WorkingDirectory
+          - MemorySize
+      - Label:
+          default: Synapse
+        Parameters:
+          - SynapseUser
+          - SynapsePassword
+      - Label:
+          default: Execution Trigger
+        Parameters:
+          - EnableSchedule
+          - Schedule
+Parameters:
+  MemorySize:
+    Type: Number
+    MinValue: 128
+    MaxValue: 10240
+    Default: 128
+    Description: >
+      The amount of memory available to the function at runtime. Increasing the function's
+      memory also increases its CPU allocation.
+  ImageUri:
+    Type: String
+    Description: >
+      The URI of the Amazon ECR) repository for the Lambda function's container image.
```

Comment:
I believe we want to pull images from the Synapse reg... is this really limited to ECR?

Reply:
hmm, from reading the docs again i think ImageUri can only pull from ECR.   Making a note to myself to  test getting an image from public dockerhub respository.

Reply:
I have tried pointing this to a test image on my public dockerhub repository (https://hub.docker.com/r/zaro0508/docker-lambda) but I cannot make it work.  Also I've checked the AWS docs again and I cannot find any references to lambda support for pulling images from dockerhub :(

Reply:
Thinking about it, it makes sense that AWS Lambda would not want to be throttled by DockerHub.  Requiring you to use an AWS registry means they don't get dragged into the pay-or-be-throttled morass.

Please check with Allaway and other key users to see if this limitation is prohibitive.  Maybe they are fine with putting their images into an AWS registry.